ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRao.java

```diff
@@ -46,9 +53,37 @@ public String getVersion() {
         }
         crac.generateValidityReport(network);
 
+        // compute maximum loop flow value F_(0,all)_MAX, and update it for each Cnec in Crac
+        SearchTreeRaoParameters searchTreeRaoParameters = parameters.getExtension(SearchTreeRaoParameters.class);
+        CracLoopFlowExtension cracLoopFlowExtension = crac.getExtension(CracLoopFlowExtension.class);
+        if (!Objects.isNull(searchTreeRaoParameters) && useLoopFlowExtension(searchTreeRaoParameters)
+            && !Objects.isNull(cracLoopFlowExtension)) {
+            //For the initial Network, compute the F_(0,all)_init
+            LoopFlowComputation initialLoopFlowComputation = new LoopFlowComputation(network, crac, cracLoopFlowExtension.getGlskProvider(), cracLoopFlowExtension.getCountriesForLoopFlow());
+            Map<String, Double> fZeroAll = initialLoopFlowComputation.calculateLoopFlows();
+            updateCnecsLoopFlowConstraint(crac, fZeroAll);
+        }
+
         // run optimisation
         RaoComputationResult result = Tree.search(network, crac, variantId, parameters).join();
         return CompletableFuture.completedFuture(result);
     }
 
+    public void updateCnecsLoopFlowConstraint(Crac crac, Map<String, Double> fZeroAll) {
+        // For each Cnec, get the maximum F_(0,all)_MAX = Math.max(F_(0,all)_init, loop flow threshold
+        crac.getCnecs().forEach(cnec -> {
```

Comment:
fZeroAll only contains the values for the preventive state cnecs if I understand the LoopFlowComputation code correctly.
This means you'll get an error when trying to do fZeroAll.get(cnec.getNetworkElement().getId()), if there's a line that's only monitored during contingencies.
If we only want to monitor loopflows on preventive cnecs, change it to crac.getCnecs(crac.getPreventiveState())

Reply:
OK